src/shared/hash.h

```diff
@@ -0,0 +1,14 @@
+#ifndef HEADER_hashing
+#define HEADER_hashing
+#include <stdio.h>
+#include <openssl/sha.h>
+#include <string.h>
+
+typedef struct { char x[SHA_DIGEST_LENGTH*2 + 1]; } sha_hexdigest;
```

Comment:
SHA_DIGEST_LENGTH is the length a buffer needs to contain a SHA1 hash, however since hashes are binary strings and I want to use them as a file name I need to convert the hash to a printable string

The easiest way is to represent it as an hexadecimal string, so using the symbols from 0 to 9 and "a" to "f", since each hexadecimal digit represents 4 bits and 1 byte is 8 bits a printable "digest" will be twice as long as the binary hash

The extra symbol is to have space for a `\0` so that I can use the functions from `string.h`